src/include/txn.i

```diff
@@ -438,11 +443,19 @@ __wt_txn_op_set_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op)
 		/*
 		 * The timestamp is in the page deleted structure for
 		 * truncates, or in the update for other operations.
+		 * Both commit timestamp and durable timestamp needs to be
+		 * updated.
 		 */
 		timestamp = op->type == WT_TXN_OP_REF_DELETE ?
 		    &op->u.ref->page_del->timestamp : &op->u.op_upd->timestamp;
-		if (*timestamp == WT_TS_NONE)
+		if (*timestamp == WT_TS_NONE) {
 			*timestamp = txn->commit_timestamp;
+
+			timestamp = op->type == WT_TXN_OP_REF_DELETE ?
+			    &op->u.ref->page_del->durable_timestamp :
+			    &op->u.op_upd->durable_timestamp;
+			*timestamp = txn->durable_timestamp;
```

Comment:
Can the original assignment to `*timestamp` be replaced with a comment here? The comment would explain that "The durable timestamp is usually the commit timestamp, except when dealing with prepared transactions."

Reply:
Ignore this one